operator/pkg/controllers/pod.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	elasticv1alpha1 "github.com/intelligent-machine-learning/easydl/operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	labelAppName         = "app"
+	labelReplicaTypeKey  = "replica-type"
+	labelReplicaIndexKey = "replica-index"
+	easydlApp            = "easydl"
+)
+
+// TaskType defines the task type (ps/worker) of a pod.
+type TaskType string
+
+// PodManager manages the lifecycle of a pod including creation, updation and deletion.
+type PodManager struct{}
+
+func newPodManager() *PodManager {
+	return &PodManager{}
+}
+
+// CreatePod creates a Pod according to a PodTemplateSpec
+func (m *PodManager) CreatePod(job *elasticv1alpha1.ElasticJob, podTemplate *corev1.PodTemplateSpec, taskType TaskType, taskIndex int) *corev1.Pod {
```

Comment:
I suggest changing "CreatePod" to "GeneratePod". In controller, "CreatePod" is more like creating a Pod in k8s cluster.

Reply:
Yeah!